tests/e2e/specs/in-game/sevens.spec.js

```diff
@@ -1487,4 +1487,45 @@ describe('Playing sevens at the end of the deck', () => {
       scrap: [Card.SEVEN_OF_CLUBS],
     });
   });
+
+  it('Cannot play seven one-off if deck is empty', () => {
+    cy.setupGameAsP1();
+    cy.loadGameFixture(1, {
+      p0Hand: [],
+      p0Points: [],
+      p0FaceCards: [],
+      p1Hand: [Card.SEVEN_OF_CLUBS, Card.ACE_OF_CLUBS],
+      p1Points: [],
+      p1FaceCards: [],
+      topCard: Card.SIX_OF_HEARTS,
+      secondCard: Card.SEVEN_OF_HEARTS,
+      deck: [Card.FIVE_OF_DIAMONDS],
+    });
+    cy.drawCardOpponent();
+    cy.get('#deck').click();
+    cy.drawCardOpponent();
+    cy.get('[data-player-hand-card=7-0]').click();
+    cy.get('[data-move-choice=oneOff').should('have.class', 'v-card--disabled');
+  });
+
+  it("Cannot play last card of deck as 7 one-off when chaining 7's", () => {
+    cy.setupGameAsP1();
+    cy.loadGameFixture(1, {
+      p0Hand: [],
+      p0Points: [],
+      p0FaceCards: [],
+      p1Hand: [Card.SEVEN_OF_CLUBS, Card.ACE_OF_CLUBS],
+      p1Points: [],
+      p1FaceCards: [],
+      topCard: Card.SIX_OF_HEARTS,
+      secondCard: Card.SEVEN_OF_HEARTS,
+      deck: [],
+    });
+    cy.drawCardOpponent();
+    cy.get('[data-player-hand-card=7-0]').click();
+    cy.get('[data-move-choice=oneOff').should('not.have.class', 'v-card--disabled').click();
+    cy.resolveOpponent();
+    cy.get('#deck').click();
```

Comment:
Don't you need to specifically click the 7 of hearts in the deck, not just the deck itself?
```suggestion
    cy.get('[data-top-card=7-2]').click();
```

Reply:
huh! This works I think because basically the top card is in the middle of the deck, but in general the data-top-card selector is better b/c it's more specific (and works when both the top and second card are revealed)